File: apps/android/app/src/main/java/com/alafteknoloji/photon/core/navigation/NavigationState.kt
package com.alafteknoloji.photon.core.navigation

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import com.alafteknoloji.photon.models.LoadedPhoto

sealed class Screen {
    data object Splash : Screen()
    data object Auth : Screen()
    data object Home : Screen()
    data class Editor(val photo: LoadedPhoto) : Screen()
    data object Settings : Screen()
    data object Membership : Screen()
}

/**
 * Observable navigation coordinator managing root screen transitions and modals.
 * 100% Parity with iOS NavigationState.swift.
 */
class NavigationState {
    var currentScreen by mutableStateOf<Screen>(Screen.Splash)
        private set

    var isMembershipPresented by mutableStateOf(false)
        private set

    fun navigateToSplash() {
        currentScreen = Screen.Splash
    }

    fun navigateToAuth() {
        currentScreen = Screen.Auth
    }

    fun navigateToHome() {
        currentScreen = Screen.Home
    }

    fun navigateToEditor(photo: LoadedPhoto) {
        currentScreen = Screen.Editor(photo)
    }

    fun navigateToSettings() {
        currentScreen = Screen.Settings
    }

    fun presentMembership() {
        isMembershipPresented = true
    }

    fun dismissMembership() {
        isMembershipPresented = false
    }
}

File: apps/ios/Photon/Core/Utilities/NavigationState.swift
//
//  NavigationState.swift
//  Photon
//
//  Created by Metin TASCI on 14.08.2026.
//

import SwiftUI

/// Core application routes matching the product flow: Splash -> Auth -> Home -> Editor.
public enum AppRoute: Hashable, Identifiable {
    case splash
    case auth
    case home
    case editor
    case settings
    
    public var id: String {
        switch self {
        case .splash: return "splash"
        case .auth: return "auth"
        case .home: return "home"
        case .editor: return "editor"
        case .settings: return "settings"
        }
    }
}

/// Central navigation coordinator state managing root transitions and navigation stacks.
@MainActor
@Observable
public final class NavigationState {
    /// Active root destination (Splash, Auth, Home, Editor)
    public var currentRoot: AppRoute = .splash
    
    /// Secondary navigation path for pushed subviews
    public var path = NavigationPath()
    
    public init() {}
    
    // MARK: - Navigation Transitions
    
    public func navigateToAuth() {
        withAnimation(.easeInOut(duration: 0.25)) {
            self.currentRoot = .auth
        }
    }
    
    public func navigateToHome() {
        withAnimation(.easeInOut(duration: 0.25)) {
            self.currentRoot = .home
        }
    }
    
    public func navigateToEditor() {
        withAnimation(.easeInOut(duration: 0.25)) {
            self.currentRoot = .editor
        }
    }
    
    public func navigateToSettings() {
        withAnimation(.easeInOut(duration: 0.25)) {
            self.currentRoot = .settings
        }
    }
    
    public func resetToSplash() {
        withAnimation(.easeInOut(duration: 0.25)) {
            self.currentRoot = .splash
            self.path = NavigationPath()
        }
    }
}
